tests/drop_counters/test_drop_counters.py

```diff
@@ -202,140 +378,586 @@ def base_verification(discard_group, pkt, ptfadapter, duthost, combined_counter,
-        l3_drops = int("".join(l3_drops.split(",")))
+def str_to_int(value):
+    """ Convert string value which can contain ',' symbols to integer value """
+    return int(value.replace(",", ""))
+
+
+def verify_drop_counters(duthost, dut_iface, get_cnt_cli_cmd, column_key):
+    """ Verify drop counter incremented on specific interface """
+    drops = get_pkt_drops(duthost, get_cnt_cli_cmd)[dut_iface][column_key]
+    drops = str_to_int(drops)
+
+    if drops != PKT_NUMBER:
+        fail_msg = "'{}' drop counter was not incremented on iface {}. DUT {} == {}; Sent == {}".format(
+            column_key, dut_iface, column_key, drops, PKT_NUMBER
+        )
+        pytest.fail(fail_msg)
 
-        if l3_drops != PKT_NUMBER:
-            fail_msg = "RX_ERR drop counter was not incremented on iface {}. DUT RX_ERR == {}; Sent pkts == {}".format(
-                dut_iface, l3_drops, PKT_NUMBER
+
+def base_verification(discard_group, pkt, ptfadapter, duthost, ptf_tx_port_id, dut_iface, l2_col_key=RX_DRP, l3_col_key=RX_ERR):
+    """
+    Base test function for verification of L2 or L3 packet drops. Verification type depends on 'discard_group' value.
+    Supported 'discard_group' values: 'L2', 'L3', 'ACL'
+    """
+    send_packets(pkt, duthost, ptfadapter, ptf_tx_port_id)
+    if discard_group == "L2":
+        verify_drop_counters(duthost, dut_iface, GET_L2_COUNTERS, l2_col_key)
+        ensure_no_l3_drops(duthost)
+    elif discard_group == "L3":
+        if COMBINED_L2L3_DROP_COUNTER:
+            verify_drop_counters(duthost, dut_iface, GET_L2_COUNTERS, l2_col_key)
+            ensure_no_l3_drops(duthost)
+        else:
+            verify_drop_counters(duthost, dut_iface, GET_L3_COUNTERS, l3_col_key)
+            ensure_no_l2_drops(duthost)
+    elif discard_group == "ACL":
+        time.sleep(ACL_COUNTERS_UPDATE_INTERVAL)
+        acl_drops = duthost.acl_facts()["ansible_facts"]["ansible_acl_facts"]["DATAACL"]["rules"]["RULE_1"]["packets_count"]
+        if acl_drops != PKT_NUMBER:
+            fail_msg = "ACL drop counter was not incremented on iface {}. DUT ACL counter == {}; Sent pkts == {}".format(
+                dut_iface, acl_drops, PKT_NUMBER
             )
             pytest.fail(fail_msg)
-
-        # Skip L2 discards verification for platform with linked L2 and L3 drop counters
-        if not combined_counter:
-            # Verify L2 drop counters were not incremented
-            intf_l2_counters = get_pkt_drops(duthost, GET_L2_COUNTERS)
-            unexpected_drops = {}
-            for iface, value in intf_l2_counters.items():
-                if int(value[L2_DISCARD_KEY]) != 0:
-                    unexpected_drops[iface] = int(value[L2_DISCARD_KEY])
-            if unexpected_drops:
-                pytest.fail("L2 'RX_DRP' was incremented for the following interfaces:\n{}".format(unexpected_drops))
+        ensure_no_l3_drops(duthost)
+        ensure_no_l2_drops(duthost)
```

Comment:
Where did the check for `COMBINED_ACL_COUNTER` go? I think it's possible for both `acl_show` and `portstat` to be incremented in such a case. Maybe it would make sense to `verify_drop_counters` in this case.

Reply:
Added ```COMBINED_ACL_COUNTER```.
Regarding ```I think it's possible for both acl_show and portstat to be incremented in such a case```
Its is not 100% sure, and it can be different in different vendors, so I think adding ```verify_drop_counters``` verification can cause some failures on some platforms.

Reply:
Sounds good to me. :)